driver-core/src/test/functional/com/mongodb/client/model/AggregatesFunctionalSpecification.groovy

```diff
@@ -1275,4 +1288,59 @@ class AggregatesFunctionalSpecification extends OperationFunctionalSpecification
         actual.size() == 1
         actual.get(0) == original[0]
     }
+
+    @IgnoreIf({ serverVersionLessThan(5, 3) })
+    def '$fill'(Bson preSortBy, String field, Bson fillStage, List<Object> expectedFieldValues) {
+        given:
+        getCollectionHelper().drop()
+        Document[] docs = [
+                new Document('partition', new Document('id', 10))
+                        .append('_id', 1)
+                        .append('field1', 1)
+                        .append('doc', new Document('field2', 1)),
+                new Document('partition', new Document('id', 10))
+                        .append('_id', 2)
+                        .append('doc', null),
+                new Document('partition', new Document('id', 20))
+                        .append('_id', 3)
+                        .append('field1', 3)
+                        .append('doc', new Document('field2', 3))]
+        getCollectionHelper().insertDocuments(docs)
+        String resultField = 'result'
+        LinkedList<Bson> stages = [
```

Comment:
Q - why linkedlist over just List?

Reply:
Indeed, I could have get by with the `List` interface. Done.